datasource/opennms-common/src/main/java/org/opennms/oce/datasource/common/inventory/script/AbstractScriptedInventory.java

```diff
@@ -0,0 +1,199 @@
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.oce.datasource.common.inventory.script;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+
+import javax.script.Invocable;
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.io.Files;
+
+/**
+ * @author smith
+ *
+ */
+public abstract class AbstractScriptedInventory {
+
+    protected static final Logger LOG = LoggerFactory.getLogger(AbstractScriptedInventory.class);
+
+    protected static final String DEFAULT_SCRIPT = "/inventory.groovy";
+
+    private boolean usingClasspathScript;
+
+    private BundleContext context;
+
+    private Invocable invocable;
+
+    private String scriptPath;
+
+    private ScriptEngineManager manager;
+
+    private ScriptEngine engine;
+
+    private long configurationTimestamp;
+
+    private long scriptFileTimestamp;
+
+    private long scriptCacheMillis;
+
+    public AbstractScriptedInventory(String scriptPath, long scriptCacheMillis, BundleContext bundleContext) {
+        if (scriptPath == null) {
+            throw new IllegalArgumentException("Null value for scriptFile.");
+        }
+
+        String script;
+        String scriptExtension;
+
+        this.scriptCacheMillis = scriptCacheMillis;
+
+        if (scriptPath.isEmpty()) {
+            // load default from classpath
+            usingClasspathScript = true;
+            URL scriptUrl = bundleContext.getBundle().getResource(DEFAULT_SCRIPT);
+            StringBuilder sb = new StringBuilder();
+            try (BufferedReader br = new BufferedReader(new InputStreamReader(scriptUrl.openStream()))) {
+                String line;
+                while ((line = br.readLine()) != null) {
+                    sb.append(line).append("\n");
+                }
+            } catch (IOException e) {
+                throw new IllegalArgumentException("Cannot find script in classpath : " + e.getMessage());
+            }
+            script = sb.toString();
+            this.scriptPath = DEFAULT_SCRIPT;
+            scriptExtension = "groovy";
+            LOG.info("Loaded inventory.groovy from the classpath");
+        } else {
+            // read the script from the file system
+            this.scriptPath = scriptPath;
+
+            File file = new File(scriptPath);
+            if (!file.canRead()) {
+                throw new IllegalStateException("Cannot read script at '" + file + "'.");
+            }
+            try {
+                byte[] fileBytes = java.nio.file.Files.readAllBytes(file.toPath());
```

Comment:
Can be done more concisely using `Files.toString` from Guava.

Reply:
done